File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, N: int) -> List[List[int]]:
        triangle = [[1]]
        for row in range(1, N):
            last = triangle[-1]
            triangle.append([1] + [x+y for x,y in zip(last, last[1:])] + [1])
        return triangle

File: 118-pascals-triangle/118-pascals-triangle.kt
class Solution {
    fun generate(numRows: Int): List<List<Int>> {
        val triangle = mutableListOf(listOf(1))
        repeat(numRows - 1) {
            val last = triangle.last()
            triangle.add(listOf(1) + last.zipWithNext(Int::plus) + listOf(1))
        }
        return triangle
    }
}